pkg/dashboard/adapter/config.go

```diff
@@ -35,6 +35,7 @@ func GenDashboardConfig(srv *server.Server) (*config.Config, error) {
 	dashboardCfg.PublicPathPrefix = cfg.Dashboard.PublicPathPrefix
 	dashboardCfg.EnableTelemetry = cfg.Dashboard.EnableTelemetry
 	dashboardCfg.EnableExperimental = cfg.Dashboard.EnableExperimental
+	dashboardCfg.DisableCustomPromAddr = cfg.Dashboard.DisableCustomPromAddr
```

Comment:
Will the config be lost when we upgrade?

Reply:
> Will the config be lost when we upgrade?

This configuration already exists in the `master` and `release-6.5` branches of tidb-dashboard and will not be lost during a normal upgrade.

Reply:
What if we set it to true in 6.5.11, then the cluster is upgraded to 7.1.x which doesn't contain this PR?

Reply:
Oh, I get what you mean. This change will be cherry-picked to all LTS branches later. (At least for now only 6.5.x customers explicitly need to configure this.)

Reply:
If an upgrade occurs during this period, this configuration will indeed lose its effect.

Reply:
After losing it, if we upgrade again to a higher version with this config, the config value will be changed to false.

Reply:
I think this is acceptable and I will doc this behavior. It is expected that this configuration will not work after upgrading to a version that does not support this feature.